### ISSUE TITLE
Dependency: use hamcrest-library instead of hamcrest-all in pom.xml

### ISSUE THREAD

**Issue by GoogleCodeExporter** [dschulten/lambdaj#79]
```
What steps will reproduce the problem?
1. The hamcrest-all dependency cannot be upgrade to 1.2.1 as hamcrest-all is 
not deployed anymore in the Maven Central Repository (see 
http://code.google.com/p/hamcrest/wiki/HamcrestDistributables#Maven-Versions)
2. hamcrest-all packages all its dependencies (expecially hamcrest-core). So if 
another lib in an application needs hamcrest-core, maven will resolve 
hamcrest-all and hamcrest-core. If the 2 dependencies have not the same 
versions, this can lead to runtime errors (NoClassDefFoundError, 
NoSuchMethodError, ...)

What is the expected output? What do you see instead?
Declare hamcrest-library 1.2.1 as dependency

What version of the product are you using? On what operating system?
lambdaj 2.3.3

Please provide any additional information below.
N/A

```

Original issue reported on code.google.com by `redfish4...@gmail.com` on 13 Dec 2011 at 10:02

**Comment by GoogleCodeExporter** [comment]
```
current workaround when using lambdaj : exclude hamcrest-all and explicitly 
declare hamcrest-library

<dependency>
  <groupId>com.googlecode.lambdaj</groupId>
  <artifactId>lambdaj</artifactId>
  <version>2.3.3</version>
  <exclusions>
    <exclusion>
      <groupId>org.hamcrest</groupId>
      <artifactId>hamcrest-all</artifactId>
    </exclusion>
  </exclusions>
</dependency>  
<dependency>
  <groupId>org.hamcrest</groupId>
  <artifactId>hamcrest-library</artifactId>
  <version>1.2.1</version>
</dependency>

```

Original comment by `redfish4...@gmail.com` on 6 Jan 2012 at 4:21

**Comment by GoogleCodeExporter** [comment]
```
Fixed in lambdaj 2.4
```

Original comment by `mario.fu...@gmail.com` on 29 Jan 2012 at 11:10
- Changed state: **Fixed**
